### DISCUSSION TITLE
Add setting to rewind on double tap of headphones play/pause button

### DISCUSSION THREAD

**Discussion by kimsey0** [PaulWoitaschek/Voice#1383] [Ideas]
Sometimes, when I listen to audio books through my headphones, my listening is interrupted, and I have to rewind. I would like to be able to do this without taking out my phone. My headphones only have one button, a play/pause button. If I click it once, Voice pauses or resumes play. If I click it twice, Android [interprets this](https://developer.android.com/reference/android/support/v4/media/session/MediaSessionCompat.Callback.html#onMediaButtonEvent(android.content.Intent)) as a command to _skip to next_, and Voice fast forwards.

I would like a setting where I can change this behavior so that Voice rewinds instead. I'll be happy to implement it and submit a pull request if @PaulWoitaschek is OK with having this setting.

**Comment by innesst** [comment]
I have the "Auto Rewind" setting set to 5 seconds (can go up to 20). If I want to go back a few seconds, I hit pause then play (and maybe a second time if I want to go back more). You do have to wait about a second between pause / play so the system doesn't interpret it as a double press.

**Comment by kimsey0** [comment]
@innesst: That's not a bad idea. I'll use that if this request isn't accepted. Thanks.

**Comment by kimsey0** [comment]
@PaulWoitaschek, what do you think about having such a setting?

**Comment by jtagcat** [comment]
This would need to be implemented 'reverse jump back and jump forward', since double press makes android give the app 'you got a signal: jump fwd'.

**Comment by kimsey0** [comment]
@PaulWoitaschek, what do you think about having a setting as described here?

**Comment by MagTun** [comment]
If you implement this feature, please keep it as a settings to enable (or not) because on my bluetooth heaphone, the double press calls the last number I called which is a very (very) usefull feature. (My headphone has 3 buttons: 1=play/pause/start/stop/call last number, 2=decrease volume/rewind, 3= increase volume/fast forward)

**Comment by wberrier** [comment]
> If you implement this feature, please keep it as a settings to enable (or not) because on my bluetooth heaphone, the double press calls the last number I called which is a very (very) usefull feature. (My headphone has 3 buttons: 1=play/pause/start/stop/call last number, 2=decrease volume/rewind, 3= increase volume/fast forward)

I'm assuming events from a bluetooth headset look different from a wired headset?

I'd also appreciate this feature.  I noticed that tapping the button press twice was skipping ahead by skip amount, when often this feature is used to rewind with audio books.

Related: #456

Although the above referenced issue is customizing the double button press behavior...

IMO, skipping forward is much less useful for an audio book app than skipping backwards.
